app/src/main/java/org/kinecosystem/kinit/view/SplashActivity.kt

```diff
@@ -1,84 +1,61 @@
-
-    private lateinit var splashViewModel: SplashViewModel
-    private val isBLackedListListener = object : OnPropertyChangedCallback() {
-        override fun onPropertyChanged(observable: Observable?, propertyId: Int) {
-            if(splashViewModel.inBlackList.get()){
-                showDialogBlackedList()
-            }
-        }
-    }
+    private var googlePlayServicesAvailable: Int? = null
+    private val splashViewModel: SplashViewModel = SplashViewModel()
 
     override fun onCreate(savedInstanceState: Bundle?) {
         KinitApplication.coreComponent.inject(this)
         super.onCreate(savedInstanceState)
-        splashViewModel = SplashViewModel(this)
-        moveToSplashScreen()
-    }
-
-    override fun moveToSplashScreen() {
-        val binding = DataBindingUtil.setContentView<SplashLayoutBinding>(this, R.layout.splash_layout)
-        binding.model = splashViewModel
-    }
-
-    override fun moveToErrorScreen() {
-        val binding = DataBindingUtil.setContentView<OnboardingErrorCreatingWalletLayoutBinding>(this,
-                R.layout.onboarding_error_creating_wallet_layout)
-        binding.model = splashViewModel
-    }
-
-    override fun openContactSupport() {
-        SupportUtil.openEmailSupport(this, userRepository)
-    }
-
-    override fun moveToMainScreen() {
-        startActivity(MainActivity.getIntent(this))
-        finish()
-    }
-
-    override fun onPause() {
-        super.onPause()
-        splashViewModel.inBlackList.removeOnPropertyChangedCallback(isBLackedListListener)
+        googlePlayServicesAvailable = GoogleApiAvailability.getInstance().isGooglePlayServicesAvailable(applicationContext)
+        setContentView(R.layout.splash_layout)
+        splashViewModel.listener = this
     }
 
     override fun onResume() {
         super.onResume()
-        checkGoogleServices()
-        splashViewModel.onResume()
-        splashViewModel.inBlackList.addOnPropertyChangedCallback(isBLackedListListener)
-        if (splashViewModel.inBlackList.get()) {
-            showDialogBlackedList()
-            splashViewModel.inBlackList.removeOnPropertyChangedCallback(isBLackedListListener)
+        if (googlePlayServicesAvailable == ConnectionResult.SUCCESS) {
+            splashViewModel.onResume()
+        } else {
+            GoogleApiAvailability
+                    .getInstance()
+                    .getErrorDialog(
+                            this,
+                            googlePlayServicesAvailable
+                                    ?: -1, SplashActivity.PLAY_SERVICES_UPDATE_REQUEST)
+                    .show()
         }
     }
 
-    private fun showDialogBlackedList() {
+    override fun moveToNextScreen() {
+        val navigator = Navigator(this@SplashActivity)
+        if (splashViewModel.userRepository.isPhoneVerificationEnabled && !splashViewModel.userRepository.isPhoneVerified) {
+            navigator.navigateTo(Navigator.Destination.TUTORIAL)
```

Comment:
else if (userRepository.isFirstTimeUser && !userRepository.isPhoneVerificationEnabled) {	
            navigator.navigateTo(Navigator.Destination.TUTORIAL)
        }

Reply:
`isFirstTimeUser` is removed. in the new flow, every user is a `FirstTimeUser` and will see tutorial before phone verification completed